docs/design/kubeadm-api.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+type MasterConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Data shared between phases. Used for defaulting
+	Master     Master     `json:"master"`
+	Networking Networking `json:"networking"`
+	CertificatesDir string `json:"certificatesDir"`
+
+	// Phases
+	Certificates     Certificates     `json:"certificates"`
```

Comment:
We might group all phases under a Phase(s) struct to make things more clear.
Here, `Certificates` stands for the first phase whose task is to create certs.
`CertificatesDir` is a param to the `Certificates` phase that says where certs should be put